content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,519 @@
+
+The first step is to check the summary of petal_width 
+
+Secondly, use the information derived to convert it to a categorical variable.
+```python
+df.petal_width.describe()
+```
+![petalwidthdescribe](petalwidthdescribe.png)
+
+Since, the operation we are going to be performing are mathematical operations and they don't work with letters or words except numbers. The Chi-square test helps in determining if there is a notable difference between the observed and normal frequencies in one or more categories, the values of petal_width must be changed into categories of zeros and ones using the 50% percentile which is the second quartile as the yardstick. There are clear [differences](https://byjus.com/maths/difference-between-percentage-and-percentile/) between percentage and percentile. As percentage is just a fraction of a hundred, the percentile is more concerned with ranks, it can be termed as the percentage of values that fall below a certain value mark. It is rather used to find the position of a value in a given set of values. For example, if a girl has the 50th percentile on an examination of 200, we can say that by scoring 200 marks, she has higher marks than 50% of the remaining class.
+
+#### How to calculate percentile.
+There are two conditions that must be satisfied.
+1. Data must be arranged in ascending order.
+2. if the index is not an integer, it must be rounded up.
+```
+np.percentile(df.petal_width, 50)
+```
+![percentile](percentile.png)
+
+ A quartile is another term-related percentile, it can be called a specified percentile, we have 1st quartile as 25% percentile, second quartile as 50%, and 3rd as 75% percentile. You can as well read this [article](https://datascienceupskill.com/statistics/percentiles-and-quartiles) to get a better grip.  
+
+```python
+def petal_cat(df):
+ if df['petal_width'] <= 1.3:
+ return 0
+ elif df['petal_width'] > 1.3:
+ return 1
+ else:
+ return 'indifferent'
+
+df['petal_width_new'] = df.apply(petal_cat, axis=1)  
+```
+The non-machine-friendly format for our *species* class has to be transformed to 0,1 and 2, which is a machine-readable format for the testing to be carried out. Therefore, with [label-encoding](https://www.section.io/engineering-education/introduction-to-scikit-learn-in-python/), a numeric value will be assigned to each of the categories in the class. To learn more about label encoding, [click](https://www.section.io/engineering-education/data-preprocessing-in-r/).
+
+```python
+def species_cat(df):
+ if df["species"] == "Iris-virginica":
+ return 0
+ elif df["species"] == "Iris-versicolor":
+ return 1
+ else:
+ return 2
+
+df["species"] = df.apply(species_cat, axis=1)
+```
+```python
+df.sample(5)
+```
+![dfsample5](dfsample5.png)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](df_new.png)
+
+Let's make use of `chic2_contigency` provided for us in `scipy.stats` library.
+
```

Comment:
I see that you've got the formatting right here but there are a lot of instances in the article where it isn't right. Please make sure to read the review comments and to apply those changes to **all occurrences in the draft**